addons/website/static/src/snippets/s_announcement_scroll/announcement_scroll.js

```diff
@@ -0,0 +1,145 @@
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+import { _t } from "@web/core/l10n/translation";
+
+
+export class AnnouncementScroll extends Interaction {
+    static selector = ".s_announcement_scroll";
+
+    dynamicContent = {
+        _window: {
+            "t-on-resize": this.debounced(this.onResize, 100),
+            "t-on-scroll": this.throttled(this.onScroll),
+        },
+        ".s_announcement_scroll_marquee_container": {
+            "t-att-style": () => ({
+                "transform": `translateX(${this.parallaxPosition}%)`
+            }),
+        },
+    }
+
+    setup() {
+        this.marqueeContainerEl = this.el.querySelector('.s_announcement_scroll_marquee_container');
+        this.marqueeItemEl = this.el.querySelector('.s_announcement_scroll_marquee_item');
+        this.marqueeContainerElWidth = this.marqueeContainerEl.offsetWidth;
+        this.marqueeItemElWidth = 0;
+        this.itemsPerContainer = 0;
+        this.scrollText = this.el.dataset.scrollcontent || _t("Write your text ");
+        this.windowHeight = window.innerHeight;
+        this.parallaxPosition = -50;
+        this.lastScrollY = window.scrollY;
+        this.scrollDelta = 0;
+        this.marqueeCloneEls = [];
+        this.currentScrollTimeoutId = null;
+    }
+
+    start() {
+        this.el.style.setProperty('--marquee-item-font-size', `${this.el.dataset.marqueeFontSize}`);
+        this.marqueeItemEl.textContent = _t(this.scrollText);
+        !this.el.dataset.scrollcontent && (this.el.dataset.scrollcontent = _t("Write your text "));
+        this.marqueeItemElWidth = this.marqueeItemEl.offsetWidth;
+        this.marqueeContainerEl.style.setProperty("--marquee-item-size", this.marqueeItemElWidth);
+        this.updateMarqueeLayout();
+        this.el.classList.toggle('s_announcement_scroll_in_viewport', this.isElementInViewport());
+        // The animation should start when the computation is done,
+        // else the first element will be already further than its clones
+        this.el.classList.add('s_announcement_scroll_ready');
+    }
+
+    /**
+     * Handles window resize events by recalculating dimensions and updating the
+     * marquee layout.
+     */
+    onResize() {
+        this.el.classList.remove('s_announcement_scroll_ready');
+        this.marqueeItemElWidth = this.marqueeItemEl.offsetWidth;
+        this.marqueeContainerElWidth = this.marqueeContainerEl.offsetWidth;
+        this.windowHeight = window.innerHeight;
```

Comment:
Seems useless: can't you just use `window.innerHeight` directly everytime? is this actually costly to read?